2363/1/triangle.rb

```diff
@@ -0,0 +1,28 @@
+# Triangle Project Code.
+
+# Triangle analyzes the lengths of the sides of a triangle
+# (represented by a, b and c) and returns the type of triangle.
+#
+# It returns:
+#   :equilateral  if all sides are equal
+#   :isosceles    if exactly 2 sides are equal
+#   :scalene      if no sides are equal
+#
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+# This method smells of :reek:UtilityFunction
+# This method smells of :reek:FeatureEnvy
+# This method smells of :reek:TooManyStatements
+def triangle(*sides)
+  sides.sort!
+  raise TriangleError if sides.any? { |side| side <= 0 } || ((sides[0] + sides[1]) <= sides[2])
```

Comment:
Here you can replace `side <= 0` expression with `.negative?` method. And such way, you can transform the whole block `{ |side| side <= 0 }` to a `Symbol` (google `ruby method shorthands` 🌞)

You can ask "For what fucking sake I should do that?" and I'll patiently answer:
* first of all, it's shorter 
* secondly, it's actually a little bit faster! Yes, it's irrelevant here, but it's cool to know, amirite? 😉

```ruby
2.5.1 :053 > Benchmark.ips do |x|
2.5.1 :054 >     x.report('block with <=')  { sides.any? { |side| side <= 0 } }
2.5.1 :055?>     x.report('block with negative?')  { sides.any? { |side| side.negative? } }
2.5.1 :056?>     x.report('Symbol#to_proc') { sides.any?(&:negative?) }
2.5.1 :057?>     x.compare!
2.5.1 :058?> end

Warming up --------------------------------------
       block with <=   151.897k i/100ms
block with negative?   134.398k i/100ms
      Symbol#to_proc   149.107k i/100ms
Calculating -------------------------------------
      block with <=             2.636M (± 4.5%) i/s -     13.215M in   5.023704s
      block with negative?      2.158M (± 4.7%) i/s -     10.886M in   5.056460s
      Symbol#to_proc            2.668M (± 5.9%) i/s -     13.420M in   5.047230s

Comparison:
      Symbol#to_proc:        2667909.7 i/s
      block with <=:         2635739.8 i/s - same-ish: difference falls within error
      block with negative?:  2157561.1 i/s - 1.24x  slower
```

Reply:
Ideally, you would like to split these validations and provide a message to these errors.

It's pretty easy thing to do:
```ruby
raise TriangleError, 'Side length cannot be less than a zero!' if ...
raise TriangleError, 'Sum of two lesser sides cannot be less that third side!' if ...
```

Reply:
You are funny:) I will remake it, give me time

Reply:
I have some troubles. I reinstall my ubuntu yesterday. And now i can create commit in new PR. Will i create new PR and close this?